tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/PartAndFilterExprRewriter.java

```diff
@@ -0,0 +1,106 @@
+package com.pingcap.tikv.expression.visitor;
+
+import com.pingcap.tikv.expression.ColumnRef;
+import com.pingcap.tikv.expression.ComparisonBinaryExpression;
+import com.pingcap.tikv.expression.ComparisonBinaryExpression.NormalizedPredicate;
+import com.pingcap.tikv.expression.Constant;
+import com.pingcap.tikv.expression.Expression;
+import com.pingcap.tikv.expression.FuncCallExpr;
+import com.pingcap.tikv.expression.FuncCallExpr.Type;
+import com.pingcap.tikv.expression.LogicalBinaryExpression;
+import com.pingcap.tikv.predicates.PredicateUtils;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * PartAndFilterExprRewriter takes partition expression as an input. Rewriting rule is based on the
+ * type of partition expression. 1. If partition expression is a columnRef, no rewriting will be
+ * performed. 2. If partition expression is year and the expression to be rewritten in the form of y
+ * < '1995-10-10' then its right hand child will be replaced with "1995". 3. If partition expression
+ * is year and the expression to be rewritten in the form of year(y) < '1995' then its left hand
+ * child will be replaced with y.
+ */
+public class PartAndFilterExprRewriter extends DefaultVisitor<Expression, Void> {
+  private Expression partExpr;
+  private Set<ColumnRef> columnRefs;
+
+  private boolean unsupportedPartFnFound;
+
+  PartAndFilterExprRewriter(Expression partExpr) {
+    Objects.requireNonNull(partExpr, "partition expression cannot be null");
+    this.partExpr = partExpr;
+    this.columnRefs = PredicateUtils.extractColumnRefFromExpression(partExpr);
+  }
+
+  private boolean isYear() {
+    return partExpr instanceof FuncCallExpr;
```

Comment:
Is this right? At least you should check enum value is year.

Reply:
Yes. I will add this check.